engine/access/subscription/block_tracker.go

```diff
@@ -0,0 +1,150 @@
+package subscription
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/counters"
+	"github.com/onflow/flow-go/module/state_synchronization"
+	"github.com/onflow/flow-go/state/protocol"
+
+	"github.com/onflow/flow-go/storage"
+)
+
+// BlockTracker is an interface for tracking blocks and handling block-related operations.
+type BlockTracker interface {
+	BaseTracker
+	// GetHighestHeight returns the highest height based on the specified block status.
+	// Only flow.BlockStatusFinalized and flow.BlockStatusSealed are allowed.
+	//
+	// Returns expected errors:
+	// - codes.InvalidArgument: If the specified block status is unsupported.
```

Comment:
To best of my understanding `BlockTracker` is a internal API that is used by trusted components. Such higher level logic needs to ensure that parameters are checked for validity. In my opinion passing an invalid enum at this point should be threaten as exception since we **_expect_** that it will be a valid value and we don't want to obfuscate the error with sentinel. In other words, if I am correct in my reasoning then receiving here an invalid enum is symptom of implementation bug not as expected input.